lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -334,18 +331,22 @@ public Object run() throws Exception {
           });
 
       LOG.log(Level.FINE, "YARN registration: register AM at \"{0}:{1}\" tracking URL \"{2}\"",
-          new Object[] {AM_REGISTRATION_HOST, AM_REGISTRATION_PORT, this.trackingUrl});
+          new Object[] {this.addressProvider.getLocalAddress(),
+              this.trackingURLProvider.getPort(),
+              this.trackingURLProvider.getTrackingUrl()});
 
       this.registration.setRegistration(this.resourceManager.registerApplicationMaster(
-          AM_REGISTRATION_HOST, AM_REGISTRATION_PORT, this.trackingUrl));
+          this.addressProvider.getLocalAddress(),
```

Comment:
TrackingURLProvider is designed for getting host:port for http server. I thought we talked and don't want to mixture it with anything in YarnContainerManager.  LocalAddressProvider can be injected in both YarnContainerManager and HttpTrackingURLProvider. It is singleton anyway. But if we want use it in  HttpTrackingURLProvider eventually, then we can inject it to it now and use it for both cases. 
Now we don't use port from HttpTrackingURLProvider, instead, register -1 as default.